dev/bots/prepare_package.dart

```diff
@@ -61,131 +66,314 @@ class ArchiveCreator {
 
   final Directory flutterRoot;
   final Directory tempDir;
-  final File outputFile;
-  final ProcessRunner _runner;
+  final Directory outputDirectory;
+  final bool _subprocessOutput;
+  final ProcessStarter _starter;
   String flutter;
   final String git = Platform.isWindows ? 'git.bat' : 'git';
   final String zip = Platform.isWindows ? '7za.exe' : 'zip';
   final String tar = Platform.isWindows ? 'tar.exe' : 'tar';
+  final Uri minGitUri = Uri.parse(MINGIT_FOR_WINDOWS_URL);
   Map<String, String> environment;
 
+  /// Performs all of the steps needed to create an archive.
+  Future<File> createArchive(String revision) async {
+    await checkoutFlutter(revision);
+    await installMinGitIfNeeded();
+    await populateCaches();
+    final File outputFile =
+        new File(path.join(outputDirectory.absolute.path, await _getArchiveName(revision)));
+    await archiveFiles(outputFile);
+    return outputFile;
+  }
+
+  Future<String> getArchiveName(String revision) async {
```

Comment:
Add a doc comment to this public member?  got slightly confused since there is also a private _getArchiveName..